galsim/config/input_powerspectrum.py

```diff
@@ -41,9 +43,45 @@ def getKwargs(self, config, base, logger):
 
         @returns kwargs, safe
         """
+        logger = galsim.config.LoggerWrapper(logger)
+
+        # If we are going to use a different rebuilding cadence than the normal once per image,
+        # then in order for this feature to work properly in a multiprocessing context,
+        # we need to have it use an rng that also updates at the same cadence as this
+        # index value.  So if we don't have an rng_num yet, make a new random_seed value
+        # that tracks this index.
+        if ('index' in config and 'rng_num' not in config and 'image' in base and
+            'random_seed' in base['image']):
+            obj_num = base.get('obj_num',None)
+            image_num = base.get('image_num',None)
+            file_num = base.get('file_num',None)
+            base['obj_num'] = base['image_num'] = base['file_num'] = 0
+            if isinstance(base['image']['random_seed'],list):
+                first_seed = galsim.config.ParseValue(base['image']['random_seed'], 0, base, int)[0]
+            else:
+                first_seed = galsim.config.ParseValue(base['image'], 'random_seed', base, int)[0]
+            rs = base['image']['random_seed']
+            if not isinstance(rs, list): rs = [rs]
+            first_seed += 31415  # An arbitrary offset to avoid the chance of unwanted correlations
```

Comment:
I don't think it's a problem for it to be hard-coded.  I just needs to be something arbitrary and non-zero.  Presumably small numbers like 1 or 2 might be problematic, but I think anything largish like this will make this an independent random number sequence.  Especially since the cadence of this sequence (typically indexed on image_num or file_num) is different from the normal rng indexing cadence (indexed on obj_num).

If the user wants, they can override this whole section by providing their own rng specification (with rng_num and making random_seed a list).  So we're not forcing the user to use this hard-coded value.  Just making it easy for people to not have to think about it too hard.